p2p/net/swarm/dial_sync.go

```diff
@@ -30,14 +37,17 @@ type dialSync struct {
 type activeDial struct {
 	refCnt int
 
-	ctx    context.Context
-	cancel func()
+	ctx         context.Context
+	cancelCause func(error)
 
 	reqch chan dialRequest
 }
 
-func (ad *activeDial) close() {
-	ad.cancel()
+func (ad *activeDial) close(err error) {
```

Comment:
Nit: it might be clearer if you just inline this function, as it's only used once.